CI/unit_tests/memory_manager/test_memory_manager.py

```diff
@@ -0,0 +1,248 @@
+        self.assertEqual(parameters["outer_scale_factor"], 2)
+        self.assertEqual(function(10, **parameters), 800)
+
+    def test_polynomial_scale_function(self):
+        """
+        Test the polynomial scale selection.
+
+        Notes
+        -----
+        Each test will check that the function parameters are return correctly and
+        that the function is called correctly and returns proper values.
+        """
+        # Test polynomial function
+        scale_function = {
+            "polynomial": {"inner_scale_factor": 2, "outer_scale_factor": 2, "order": 3}
+        }
+        function, parameters = self.memory_manager._select_scale_function(
+            scale_function
+        )
+        self.assertEqual(parameters["inner_scale_factor"], 2)
+        self.assertEqual(parameters["outer_scale_factor"], 2)
+        self.assertEqual(parameters["order"], 3)
+        self.assertEqual(function(10, **parameters), 16000)
+
+    def test_get_batch_size(self):
+        """
+        Test the get_batch_size method.
+
+        Returns
+        -------
+
+        """
+        # Test the exception catch.
+        self.memory_manager.data_path = None
+        self.assertRaises(ValueError, self.memory_manager.get_batch_size)
+
+        # Test correct returns for 1 batch
+        self.memory_manager.database = TestDatabase(data_size=500, rows=10, columns=10)
+        self.memory_manager.data_path = ["Test/Path"]
+        self.memory_manager.memory_fraction = 0.5
+        self.memory_manager.machine_properties["memory"] = 50000
+        batch_size, number_of_batches, remainder = self.memory_manager.get_batch_size(
+            system=False
+        )
+        self.assertEqual(batch_size, 10)
+        self.assertEqual(number_of_batches, 1)
+        self.assertEqual(remainder, 0)
+
+        # Test correct returns for N batches
+        self.memory_manager.database = TestDatabase(data_size=500, rows=11, columns=13)
+        self.memory_manager.data_path = ["Test/Path"]
+        self.memory_manager.memory_fraction = 1.0
+        self.memory_manager.machine_properties["memory"] = 50
+        batch_size, number_of_batches, remainder = self.memory_manager.get_batch_size(
+            system=False
+        )
+        self.assertEqual(batch_size, 1)
+        self.assertEqual(number_of_batches, 13)
+        self.assertEqual(remainder, 0)
+
+    def test_hdf5_load_time(self):
+        """
+        Test the hdf5_load_time method.
+
+        Returns
+        -------
+        Tests that the method returns the correct load time.
+        """
+        data = self.memory_manager.hdf5_load_time(10)
+        self.assertEqual(data, np.log(10))
+
+    def test_get_optimal_batch_size(self):
+        """
+        Test the get_optimal_batch_size method.
+
+        Returns
+        -------
+        Test that this method returns the expected value. Currently this is just
+        the same value that is passed to it.
+        """
```

Comment:
will this change? Maybe add a TODO

Reply:
The test is not where the issue is. It is in the actual memory management. It won't be changed until we have a better approach to the memory management.